### PR TITLE
feat: account password policy attachment

### PR DESCRIPTION
**Pull request by sfc-gh-ngaberel**
Adds support for attaching a password policy to the current account.

Due to limitations of the underlying API, this feature only supports setting the password policy of the **current account** (ie. the one from the `provider snowflake {}` block). To set the password policy of a different account, configure an aliased provider for that account.

## Test Plan
* [X] acceptance tests

## References

* Implements #1746 
* Depends on #1822

### CODE REVIEW

File: pkg/resources/account_password_policy_attachment.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var accountPasswordPolicyAttachmentSchema = map[string]*schema.Schema{
+	"password_policy": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Qualified name (`\"db\".\"schema\".\"policy_name\"`) of the password policy to apply to the current account.",
```
Comment (sfc-gh-swinkler) [inline pkg/resources/account_password_policy_attachment.go:18]: if we are going to do this, we should probably make data source(s) to help make it easier for people to make fully qualified names from db / schema / object names

File: pkg/resources/account_password_policy_attachment.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var accountPasswordPolicyAttachmentSchema = map[string]*schema.Schema{
```
Comment (sfc-gh-swinkler) [inline pkg/resources/account_password_policy_attachment.go:13]: should this be renamed to "password_policy_account_attachment"?

Comment (sfc-gh-ngaberel) [reply pkg/resources/account_password_policy_attachment.go:13]: password policies can be attached to either accounts or users, so I figured we'd have a separate resource later user_password_policy_attachment

File: pkg/resources/account_password_policy_attachment.go (hunk 1 above)
Comment (sfc-gh-ngaberel) [reply pkg/resources/account_password_policy_attachment.go:18]: I added the qualified name as an attribute of the password policy resource, but the data source is a good idea too.

File: docs/resources/password_policy.md [hunk 3]
```diff
@@ -39,5 +39,6 @@ A password policy specifies the requirements that must be met to create and rese
 ### Read-Only
 
 - `id` (String) The ID of this resource.
+- `qualified_name` (String) The qualified name for the password policy.
```
[the remark this answers is not in the document]
Comment (sfc-gh-ngaberel) [reply docs/resources/password_policy.md:42]: This matches the attribute on masking policies, but we can change them.

**Pull request merged**